io/PcdWriter.cpp

```diff
@@ -271,35 +280,96 @@ void PcdWriter::write(const PointViewPtr view)
         point.setPointId(idx);
         if (m_compression_string == "ascii")
         {
+            *m_ostream << std::fixed;
             for (auto di = m_dims.begin(); di != m_dims.end(); ++di)
-                *m_ostream << std::fixed << std::setprecision(di->m_precision)
-                           << point.getFieldAs<float>(di->m_field.m_id) << " ";
+            {
```

Comment:
This particular change is aimed at outputting the field according to its dimension specification. We may default to double or float at a given precision, but if the user specified something else via `order` it would have been disregarded in the ASCII output.

Reply:
For example, on master, the command

```
$ pdal translate ../test/data/las/utm17.las utm17.pcd \
    --writers.pcd.compression="ascii" -v 5 \
    --writers.pcd.order="X=Unsigned32:2,Y=Double,Z" \
    --writers.pcd.keep_unspecified=false
```

would give

```bash
$ cat utm17.pcd 
VERSION 0.7
FIELDS x y z
SIZE 4 8 8
TYPE U F F
COUNT 1 1 1
WIDTH 10
HEIGHT 1
VIEWPOINT 0.0 0.0 0.0 1.0 0.0 0.0 0.0
POINTS 10
DATA ascii
289814.16 4320978 170.76
```

With this PR, it would correctly yield

```bash
$ cat utm17.pcd 
VERSION 0.7
FIELDS x y z
SIZE 4 8 4
TYPE U F F
COUNT 1 1 1
WIDTH 10
HEIGHT 1
VIEWPOINT 0.0 0.0 0.0 1.0 0.0 0.0 0.0
POINTS 10
DATA ascii
289814 4320978.61 170.76
```

Reply:
OK, but I think that there are only two cases - integer (signed or unsigned) and floating, right? You're setting precision based on the option, rather than the type, so you could cut out a bunch of cases if I'm not mistaken. But it's fine either way. Up to you.

Reply:
Okay, so this is true. I'll clean it up.